File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/domain/credential/OtpCredential.java
package com.example.chat.auth.server.core.domain.credential;

import com.example.chat.auth.server.core.domain.AuthLevel;
import com.example.chat.auth.server.core.domain.Credential;
import com.example.chat.auth.server.core.domain.CredentialType;
import lombok.Getter;

/**
 * OTP 자격증명
 */
@Getter
public final class OtpCredential extends Credential {
    private final String code;
    private final String deliveryMethod;

    public OtpCredential(String code, String deliveryMethod) {
        super(CredentialType.OTP, false);
        this.code = code;
        this.deliveryMethod = deliveryMethod;
    }

    @Override
    public AuthLevel minAuthLevel() {
        return AuthLevel.LOW;
    }
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/domain/credential/OtpCredential.kt

package com.example.chat.auth.server.core.domain.credential

import com.example.chat.auth.server.core.domain.AuthLevel
import com.example.chat.auth.server.core.domain.Credential
import com.example.chat.auth.server.core.domain.CredentialType

/**
 * OTP 자격증명
 * - SMS, Email, Authenticator App으로 전달되는 일회용 코드
 */
class OtpCredential(
    val code: String,
    val deliveryMethod: String
) : Credential(CredentialType.OTP, false) {
    override fun minAuthLevel(): AuthLevel = AuthLevel.LOW
}
